src/base.py

```diff
@@ -152,11 +147,213 @@ def __init__(self, **kwargs):
             raise ValueError(f"dim = {self.dim} not supported")
         
         # Compute the bounding box indices for boundary conditions
-        self.boundingBoxIndices = self.bounding_box_indices()
+        self.boundingBoxIndices= self.bounding_box_indices()
         # Create boundary data for the simulation
         self._create_boundary_data()
         self.force = self.get_force()
 
+    @property
+    def lattice(self):
+        return self._lattice
+
+    @lattice.setter
+    def lattice(self, value):
+        if value is None:
+            raise ValueError("Lattice type must be provided.")
+        if self.nz == 0 and not isinstance(value, LatticeD2Q9):
```

Comment:
This does not "assume" any default lattice for 2D scenarios. We can still assume D2Q9 for 2D flows and D3Q27 for 3D flows for ALL cases as the default setting when no lattice input is defined by the user.